index.js

```diff
@@ -315,8 +315,12 @@ function normalizeArguments(url, opts) {
 			throw new TypeError('options.body must be a ReadableStream, string, Buffer or plain Object');
 		}
 
-		if ((opts.form || opts.json) && !isPlainObj(body)) {
```

Comment:
I'd prefer if we kept this joint, it's simpler. Let's discuss it over in the issue.